test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/LocalClusterFactory.java

```diff
@@ -673,6 +673,7 @@ private Map<String, String> getEnvironmentVariables() {
                 + " "
                 + featureFlagProperties
                 + systemProperties
+                + ' '
```

Comment:
Good sleuthing! Do we have the same potential issue between all these string values? Maybe we should build this string in a similar way to previous ones, with a stream and joiner.

Reply:
I agree, a stream and joiner would seem more robust.

Reply:
Yeah, sorry about that. It was the end of the day and I wanted to get something out the door, so I ignored the screaming voice inside me telling me how dirty the fix was.
🙀 
Updated in d6ea5b3